tests/test_router.py

```diff
@@ -53,6 +55,45 @@ def test_other(self):
         from django.contrib.auth import get_user_model
         self.assertEqual(self.sut.db_for_read(model=get_user_model()), None)
 
+    def test_router_gets_filter_hints(self):
+        TestModel.objects.create(user_pk=self.user.pk)
+
+        lookups_to_find = {'exact_lookups': {'user_pk': self.user.pk}}
+
+        with patch.object(ShardedRouter, 'db_for_read') as read_route_function:
+            read_route_function.return_value = 'app_shard_001'
+            test_models = list(TestModel.objects.filter(user_pk=self.user.pk))
+            self.assertIn(lookups_to_find, read_route_function.call_args)
+
+    def test_queryset_router_filter_returns_existing_objects(self):
+        # Cant for sure know what order tests ran in, lets make sure the delete function works in this test
+        TestModel.objects.filter(user_pk=self.user.pk).delete()
+        for i in range(1, 11):
+            test_model_obj = TestModel.objects.create(user_pk=self.user.pk, random_string="%s" % i)
+            self.assertIn(test_model_obj._state.db, ['app_shard_001', 'app_shard_002'])
+
+        test_models = TestModel.objects.filter(user_pk=self.user.pk)
+        self.assertEqual(len(test_models), 10)
+
+        from django.contrib.auth import get_user_model
+        new_user = get_user_model().objects.create_user(username='username2', password='pwassword', email='test2@example.com')
+        for i in range(1, 21):
+            TestModel.objects.create(user_pk=new_user.pk, random_string="%s" % i)
+
+        test_models = TestModel.objects.filter(user_pk=new_user.pk)
+        self.assertEqual(len(test_models), 20)
+
+    def test_queryset_router_filter_with_aggregates(self):
+        # Cant for sure know what order tests ran in, lets make sure the delete function works in this test
```

Comment:
This is unnecessary, set multi_db = True on the base test class object in the test settings. That should properly cleanup.

do this:

I forgot
```python
from django.test import TestCase
TestCase.maxDiff = None
TestCase.multi_db = True
```